### ISSUE TITLE
Doesn't work in Safari 10

### ISSUE THREAD

**Issue by tomtheisen** [tomtheisen/stax#7]
The Stax interpreter doesn't seem to work at all in Safari: Permalinks don't load any code, the buttons basically do nothing and it is overall unresponsive.

**Comment by tomtheisen** [comment]
If any passers-by happens upon this, I'd like to ask a favor.

1. Use Safari.
2. Navigate to https://staxlang.xyz/
3. Check the javascript developer console.
4. Post the contents here.  It's supposed to be empty, but I suspect there will be some kind of error reported there.

**Comment by danielmayorga** [comment]
I was using Safari Version 11.0.3 (13604.5.6) on my desktop and Mobile Safari version 11.0.
I ran the [Binary Multiplication Test](https://github.com/tomtheisen/stax/blob/master/testspecs/golf/BinaryMultiplicationTest.staxtest) and the [Basic Ascii Tables Test](https://github.com/tomtheisen/stax/blob/e6a7b46e859aa4ab11f5cf126f379efb68721209/testspecs/golf/BasicAsciiTables.staxtest)
Didn't see anything in the console logs. Everything seemed pretty responsive.

**Comment by tomtheisen** [comment]
Thanks.  This was a bug reported to me in a stack exchange chat room.  I haven't observed the behavior myself.

**Comment by tomtheisen** [comment]
According to https://chat.stackexchange.com/transcript/message/43175343#43175343, the problem is reproducible under Safari 10.0.2

> SyntaxError: Cannot declare a let variable twice: 'e'.

**Comment by tomtheisen** [comment]
Looks like a fixed Safari 10 bug.  I probably won't try to fix this.

More info: https://github.com/mishoo/UglifyJS2/issues/1753

**Comment by tomtheisen** [comment]
Possibly fixed in fbfeabec25d78f2e51353a836c21adbe721e34f0.

Don't have a way to test atm.

**Comment by danielmayorga** [comment]
The version of uglify that you were using didn't support safari10 as a command (https://github.com/webpack-contrib/uglifyjs-webpack-plugin/releases/tag/v1.2.0)
This should resolve the issue. I also checked the bundle.js file to make sure the issue didn't occur.
#16

**Comment by LoganDark** [comment]
`let a = 3
let a = 2`

Don't do that. Do this:

`let a = 3
a = 2`

That is my guess for what the error means.

**Comment by tomtheisen** [comment]
There's no need to guess.  This is a case of Uglify producing output like this:

    function(e) {
        for (let e of a) {
            // some other code
        }
    }

This is legal ES6, and there's not any code like that in this repo.  It's output from Uglify.

**Comment by LoganDark** [comment]
Apparently it's not legal in Safari(?)

**Comment by tomtheisen** [comment]
@LoganDark:  Correct.  It least it's not legal in Safari 10, insofar as there was a bug that didn't interpret it correctly.  It appears to have been fixed in Safari 11.  This stuff is kind of hard for me diagnose though, since I don't have direct access to Safari, especially old versions.

**Comment by LoganDark** [comment]
@tomtheisen I can be your tester. I'm running macOS, I can grab an older version of Safari.

**Comment by tomtheisen** [comment]
I recently discovered all the webpack configurations I was applying were not affecting the output.  That explains a lot, but also creates more mystery.  According to my original webpack config, minification was apparently happening twice.  In the first pass, the bad variables were introduced.  All the configuration I was doing was for the second pass.

Of course, two passes of minification is stupid, so I took out the second one.  At this point though, I can't figure out how to configure the first (implicit?) minification or even turn it off.  According to webpack-cli docs, you can use `--optimize-minimize` to turn on or off minification, but I can't get it to do anything at all.

So basically, I just can't figure out webpack.  PRs welcome.

**Comment by danielmayorga** [comment]
I ended manually configuring the [optimization](https://webpack.js.org/configuration/optimization/) property in the webpack config files.

I created two webpack config files: dev and prod.

Prod has the optimization configuration for [minimize](https://webpack.js.org/configuration/optimization/#optimizationminimizer) set to true and the [minimizer.](https://webpack.js.org/configuration/optimization/#optimizationminimizer) set to the UglifyJS plugin. I compared the minified files with and without the safari10 property and it looked like it addressed the original issue.

Dev has the optmization configuration for minimize set to false. I verified that it doesn't minify the bundle.js file. Don't know your workflow so 

I also made changes to the npm script because I was afraid of global package version mismatches and wanted contributors to avoid calling webpack directly and just call the scripts. 

PR - #32 

I don't have a copy of safari10, so I couldn't test it aside from comparing the minified file for variables that avoid the safari issue. Maybe I'll plug in my imac and see if I can download version 10 on the weekend.
